src/rigid_optimization/sdf_2_sdf_optimizer2d.cpp

```diff
@@ -0,0 +1,151 @@
+{
+};
+
+Sdf2SdfOptimizer2d::VerbosityParameters::VerbosityParameters(
+        bool print_iteration_max_warp_update,
+        bool print_iteration_energy) :
+            print_iteration_max_warp_update(print_iteration_max_warp_update),
+            print_iteration_energy(print_iteration_energy),
+            print_per_iteration_info(
+                    print_iteration_max_warp_update ||
+                    print_iteration_energy
+            )
+{
+};
+
+Sdf2SdfOptimizer2d::TSDFGenerationParameters::TSDFGenerationParameters(
+        float depth_unit_ratio,
+        eig::Matrix3f camera_intrinsic_matrix,
+        eig::Matrix4f camera_pose,
+        eig::Vector3i array_offset,
+        int field_size,
+        float voxel_size,
+        int narrow_band_width_voxels) :
+            depth_unit_ratio(depth_unit_ratio),
+            camera_intrinsic_matrix(camera_intrinsic_matrix),
+            camera_pose(camera_pose),
+            array_offset(array_offset),
+            field_size(field_size),
+            voxel_size(voxel_size),
+            narrow_band_width_voxels(narrow_band_width_voxels)
+{
+};
+
+eig::Vector3f Sdf2SdfOptimizer2d::optimize(int image_y_coordinate,
+        const eig::Matrix<unsigned short, eig::Dynamic, eig::Dynamic>& canonical_depth_image,
+        const eig::Matrix<unsigned short, eig::Dynamic, eig::Dynamic>& live_depth_image,
+        const Sdf2SdfOptimizer2d::TSDFGenerationParameters tsdf_generation_parameters,
+        float eta = 0.01f) {
+
+    eig::MatrixXf canonical_field = tsdf::generate_TSDF_2D(image_y_coordinate,
+                                                           canonical_depth_image,
+                                                           tsdf_generation_parameters.depth_unit_ratio,
+                                                           tsdf_generation_parameters.camera_intrinsic_matrix,
+                                                           tsdf_generation_parameters.camera_pose,
+                                                           tsdf_generation_parameters.array_offset,
+                                                           tsdf_generation_parameters.field_size,
+                                                           tsdf_generation_parameters.voxel_size,
+                                                           tsdf_generation_parameters.narrow_band_width_voxels);
+
+    eig::MatrixXf canonical_weight = canonical_field.replicate(1, 1);
+    for (int i=0; i < canonical_weight.rows(); ++i) { // Determine weight based on thickness
+        for (int j=0; j < canonical_weight.cols(); ++j){
+            canonical_weight(i, j) = (canonical_field(i, j) <= -eta) ? 0.0f : 1.0f;
+        }
+    }
+
+    eig::Vector3f twist = eig::Vector3f(0.0f, 0.0f, 0.0f);
+
+    for (int iteration_count=0; iteration_count<maximum_iteration_count; ++iteration_count) {
+        eig::Matrix3f matrix_A = eig::Matrix3f::Zero();
+        eig::Vector3f vector_b = eig::Vector3f::Zero();
```

Comment:
Are there any names that are more descriptive than "matrix_A",  "vector_b", and "twist_star"?

Reply:
The A and b are from the paper, they are used to calculate the optimal twist, I change the twist_star to optimal_twist.